docs/testsuite_how_to.md

```diff
@@ -0,0 +1,204 @@
+        }
+      }
+    }
+    
+    ```
+2. Re-generate the .restspec.json and .snapshot.json files:
+
+    ```
+    gradle publishRestIdl
+    gradle publishRestSnapshot
+    ```
+
+3. Run the test server using:
+
+    ```
+    gradle JettyRunWar
+    ```
+    This will use the new restspecs to generate or update the appropriate request builders, which are used to make the
+    requests in the following step.
+
+4. Update RequestResponseTestCases in the language-specific suites of client-testsuite by adding a new request and
+  test name to the map of Rest.li requests to be tested. 
+
+   For the java implementation, this is done by modifying the builtRequests map in the buildRequests() function,
+   in client-testsuite/src/test/java/com/linkedin/pegasus/testsuite/RequestResponseTestCases.java.
+   ```java
+    builtRequests.put("collectionReturnEntity-create", new CollectionReturnEntityRequestBuilders(_options).create().input(testMessage).build());
+    ```
+
+5. Re-generate the request and response files.  Files will be written to the "requests" and "responses" directories:
+
+    ```
+    gradle generateRequestAndResponseFiles
+    ```
+    Note that Java's requestBuilders are generating the request files with the desired output. 
+    These flat files can be used to test the other implementations for well-formed requests.
+
+6. Update the "wireProtocolTestData" entry of manifest.json to include test data references to all the files you've added.
+ 
+   If overriding a new method for an existing resource: 
+   * Find name of your resource in the wireProtocolTestData list. Under "operations", add test for the new method that you added
+   to RequestResponseTestCases in Step 4. The new operation test should look something like this:
+ 
+   ```
+   { "name":"collection-get", "method": "get", "request": "requests/collection-get.req", "response": "responses/collection-get.res", "status": 200 },
+   ```
+   where the test name is usually the resource and method. "status" is the expected response status.
+   
+   If adding a new resource:
+    * Create a new entry for your resource in wireProtocolTestData list. This should follow the 
+    general format:
+    ```json
+      "wireProtocolTestData": [
+      { 
+        "name": "collectionReturnEntity",
+        "restspec": "restspecs/testsuite.collectionReturnEntity.restspec.json",
+        "snapshot": "snapshots/testsuite.collectionReturnEntity.snapshot.json",
+        "operations": [
+        { "name":"collectionReturnEntity-create", "method": "create", "request": "requests/collectionReturnEntity-create.req", "response": "responses/collectionReturnEntity-create.res", "status": 201 }
+       ] 
+      },
+      ...
+     ]
+     ```
+    * For the list of operation tests, follow the instructions for overriding a new method for an existing resource.
+     
+7. Now that the test suite spec includes new responses, you need to update the manual assertion tests in the
```

Comment:
Make sure that it is formatted well in your preview reader, not sure why, my preview reader always not shows numbered list 7 for this.